src/main.ts

```diff
@@ -47,7 +48,7 @@ const tryRedirect = async () => {
     setMessage('Not found: No Endorsement');
   } else {
     try {
-      const href = `https://${endorsedUI.value}.ipfs.cf-ipfs.com`;
+      const href = `https://${endorsedUI.value}.ipfs.cf-ipfs.com/?${walletParam}`;
```

Comment:
Nit: Move the "?" into the template literal so it doesn't show up unless "wallet=main" does